src/lib/modules/deployment/contract_deployer.js

```diff
@@ -319,6 +318,8 @@ class ContractDeployer {
               return next(null, receipt);
             });
           });
+        }, hash => {
+          self.logFunction(contract)(__("deploying") + " " + contract.className.bold.cyan + " " + __("with").green + " " + contract.gas + " " + __("gas at the price of").green + " " + contract.gasPrice + " " + __("Wei, estimated cost:").green + " " + estimatedCost + " Wei".green + " (txHash: " + hash.bold.cyan + ")");
```

Comment:
Notice that this callback, by no means, controls the flow of the outer `async.waterfall()`. It's really just a function that receives a signal.

A problem with this API is that it might be easily overseen (also due to its position in the method signature of `secureSend()`).

Would be nice if we could refactor this API in the future (based on observables?) where we could do sth. like:

```
deployment = deployService.deployContractFromObject(obj);
deployment.txHashChange().subscribe(hash => ...);
deployment.done().subscribe(receipt => ...);
```

Another way to handle this would be to have a protocol that emits a message signal:

```
deployment$ = deployService.deployContractFromObject(obj);
txHash$ = deployment$.pipe(filter(msgAndPayload => msgAndPayload.type === MessageType.TxHash));
receipt$ = deployment$.pipe(filter(msgAndPayload => msgAndPayload.type === MessageType.Receipt));

txHash$.subscribe(hash => /* log deployment message */);
receipt$.subscribe(receipt => /* deployment done */);
```

Food for thought.